agent/hcp/deps.go

```diff
@@ -23,5 +30,71 @@ func NewDeps(cfg config.CloudConfig, logger hclog.Logger) (d Deps, err error) {
 	}
 
 	d.Provider, err = scada.New(cfg, logger.Named("hcp.scada"))
+	if err != nil {
+		return
+	}
+
+	d.SinkOpts = sinkOpts(&cfg, d.Client, logger)
+
 	return
 }
+
+// setupSink provides OTELSink configuration to initialize a Go Metrics sink,
+// only if the server is registered with the management plane (CCM).
+// This step should not block server initialization, so errors are logged, but not returned.
+func sinkOpts(cfg hcpclient.CloudConfig, client hcpclient.Client, logger hclog.Logger) *telemetry.OTELSinkOpts {
+	ctx := context.Background()
```

Comment:
Ideally we'd pass in the parent context but I'd have to modify two functions call up 😓 to pass in the parent context... I think this is fine but adds changes to external consul core code...